### PR TITLE
Fixes for Python 3

### PR DESCRIPTION
**Pull request by nphilipp**
Use the `print` function, unicode literals and `range()` instead of `xrange()`.

### CODE REVIEW

File: generate.py [hunk 1]
```diff
@@ -38,16 +40,16 @@ def writefile(filename, s):
...
-        print 'Writing', filename
+        print('Writing', filename)
         open(filename, 'w').write(s)
 
 
 def generate_enum(enum_name, enum_prefix, count_name, name_list, value_list):
     # Can only generate an enum which starts at 0, and where each value is 1 more than the former
     assert len(name_list) == len(value_list)
-    assert value_list == list(xrange(0, len(value_list)))
+    assert value_list == list(range(0, len(value_list)))
```
Comment (Jehan) [inline generate.py]: Doesn't there need to have a `from builtins import range`?
Taken from: http://python-future.org/compatible_idioms.html#xrange
But maybe this import is unneeded. I have not tested. :-)

**Comment by Jehan** [comment]
Hi,

I reviewed this. It all looks good and sane to me. Of course there is a part on a scons file which will be deleted soon anyway. But all the rest seems good to me.

And I have just a comment on the range/xrange stuff above.
Apart from this, I say this could be merged.

File: generate.py (hunk 1 above)
Comment (nphilipp) [inline generate.py]: Given that this gives an import error here (on Python 2), I don't think we should use it ;):

```
Python 2.7.11 (default, Mar 31 2016, 20:46:51) 
[GCC 5.3.1 20151207 (Red Hat 5.3.1-2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from builtins import range
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named builtins
>>>
```

Both Python 2 and 3 have a `range()` function in the global namespace, the Python 2 version returns a list and the  Python 3 version returns a range iterator. Casting either to a list will do the right thing here, with only slight overhead.

Comment (QuLogic) [inline generate.py]: [Using `builtins` requires that you install `future`, too.](http://python-future.org/reference.html?highlight=builtins#module-future.types)

But as stated, there's generally only a small overhead on Python 2.

**Comment by achadwick** [comment]
[libMyPaint has gone over to autotools now](http://mypaint.org/blog/2016/05/20/libmypaint-now-uses-autotools/). Would it be possible to update this PR so that the conflicts are resolved please?

**Comment by QuLogic** [comment]
I think almost all of this was done in #21 or makes no difference, so this can be closed.

**Pull request closed without merging**